ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2362,6 +2362,104 @@
+        "version": "8.11.3"
+      },
+      {
+        "name": "NPM",
+        "version": "5.6.0"
+      },
+      {
+        "name": "Postgres",
+        "version": "9.6"
+      }
+    ],
+    "workspaceConfig": {
+      "defaultEnv": "default",
+      "environments": {
+        "default": {
+          "machines": {
+            "database": {
+              "attributes": {
+                "memoryLimitBytes": "547483648"
+              },
+              "servers": {
+                "postgres": {
+                  "attributes": {},
+                  "port": "5432",
+                  "protocol": "tcp"
+                }
+              },
+              "volumes": {},
+              "installers": [
+                "org.eclipse.che.exec",
+                "org.eclipse.che.terminal"
+              ],
+              "env": {
+                "POSTGRESQL_ADMIN_PASSWORD": "adminpassword",
+                "POSTGRESQL_DATABASE": "my_data",
+                "POSTGRESQL_USER": "user",
+                "POSTGRESQL_PASSWORD": "password"
+              }
+            },
+            "nodejs": {
+              "attributes": {
+                "memoryLimitBytes": "1147483648"
+              },
+              "servers": {
+                "node": {
+                  "attributes": {},
+                  "port": "8080",
+                  "protocol": "http"
+                }
+              },
+              "volumes": {},
+              "installers": [
+                "org.eclipse.che.exec",
+                "org.eclipse.che.terminal",
+                "org.eclipse.che.ls.js-ts",
+                "org.eclipse.che.ws-agent"
+              ],
+              "env": {}
+            }
+          },
+          "recipe": {
+            "type": "openshift",
+            "content": "kind: List\nitems:\n - \n  apiVersion: v1\n  kind: Pod\n  metadata:\n   name: nodejs-postgres-pod\n   annotations:\n    org.eclipse.che.container.nodejs.machine_name: nodejs\n    org.eclipse.che.container.database.machine_name: database\n  spec:\n   containers:\n    - \n     image: quay.io/openshiftio/che-centos-nodejs\n     name: nodejs\n     ports:\n      - \n       containerPort: 8080\n       protocol: TCP\n     resources:\n      limits:\n       memory: 1024Mi\n    - \n     image: centos/postgresql-96-centos7\n     name: database\n     ports:\n      - \n       containerPort: 5432\n       protocol: TCP\n     resources:\n      limits:\n       memory: 512Mi\n     env:\n      - \n       name: POSTGRESQL_USER\n       value: user\n      - \n       name: POSTGRESQL_PASSWORD\n       value: password\n      - \n       name: POSTGRESQL_DATABASE\n       value: my_data\n      - \n       name: POSTGRESQL_ADMIN_PASSWORD\n       value: adminpassword\n     volumeMounts:\n      - \n       mountPath: /var/lib/pgsql/data\n       name: claim-che-workspace\n   volumes:\n    - \n     name: claim-che-workspace\n     persistentVolumeClaim:\n      claimName: claim-che-workspace\n",
```

Comment:
in the downstream PR [1] there were a good summary from @amisevsk that mounting volume `/var/lib/pgsql/data` to `claim-che-workspace`  would ruin other workspace data. Other than that it makes the stack dependent on the PVC strategy. @jarifibrahim has a suggestion [1] to create custom postgres images in `che/dockerfiles` which will not mount the `/var/lib/pgsql/data` volume which sounds relevant for me. @benoitf @eivantsov WDYT ? should we have custom postres image in `che/dockerfiles` for now ?


[1] https://github.com/redhat-developer/rh-che/pull/1086#issuecomment-443707977
[2] https://github.com/redhat-developer/rh-che/pull/1086#issuecomment-444020553

Reply:
Why can't we add a volume to postgres machine? We do add volumes for Java machines this way. Depending on the pvc strategy wither a subpath in a shared volume will be created or a new pvc bound to a new pv will be there.

Or maybe I am missing smth important...

Reply:
@eivantsov oh, great point! @jarifibrahim could you please try adding the volume [1] to postgres machine [2] ? Would be great to test if this approach would work on `che.openshift.io` 

[1] https://www.eclipse.org/che/docs/che-6/volumes.html
[2] https://github.com/eclipse/che/pull/12104/files#diff-be4e3a94b98c39160862773dcffad022R2406

Reply:
@ibuziuk Sure. I'll add a volume [here](https://github.com/eclipse/che/pull/12104/files#diff-be4e3a94b98c39160862773dcffad022R2406) but should I remove it from the recipe? or should add the volume declaration to the recipe and the machine?

Reply:
@jarifibrahim yes, we should try to remove volume from recipe and add it only to machine definition.

Reply:
@ibuziuk I've changed it in https://github.com/eclipse/che/pull/12104/commits/b1d56e7f326bdda6c1f9c8dae7d355fd415d272f